lib/class-wp-rest-plugins-controller.php

```diff
@@ -0,0 +1,822 @@
+			return $filesystem_available;
+		}
+
+		$api = plugins_api(
+			'plugin_information',
+			array(
+				'slug'   => $slug,
+				'fields' => array(
+					'sections' => false,
+				),
+			)
+		);
+
+		if ( is_wp_error( $api ) ) {
+			if ( false !== strpos( $api->get_error_message(), 'Plugin not found.' ) ) {
+				$api->add_data( array( 'status' => 404 ) );
+			} else {
+				$api->add_data( array( 'status' => 500 ) );
+			}
+
+			return $api;
+		}
+
+		$skin     = new WP_Ajax_Upgrader_Skin();
+		$upgrader = new Plugin_Upgrader( $skin );
+
+		$result = $upgrader->install( $api->download_link );
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		// This should be the same as $result above.
+		if ( is_wp_error( $skin->result ) ) {
+			return $skin->result;
+		}
+
+		if ( $skin->get_errors()->has_errors() ) {
+			return $skin->get_errors();
+		}
+
+		if ( is_null( $result ) ) {
+			global $wp_filesystem;
+			// Pass through the error from WP_Filesystem if one was raised.
+			if ( $wp_filesystem instanceof WP_Filesystem_Base && is_wp_error( $wp_filesystem->errors ) && $wp_filesystem->errors->has_errors() ) {
+				return new WP_Error( 'unable_to_connect_to_filesystem', $wp_filesystem->errors->get_error_message(), array( 'status' => 500 ) );
+			}
+
+			return new WP_Error( 'unable_to_connect_to_filesystem', __( 'Unable to connect to the filesystem. Please confirm your credentials.', 'gutenberg' ), array( 'status' => 500 ) );
+		}
+
+		$file = $upgrader->plugin_info();
+
+		if ( ! $file ) {
+			return new WP_Error( 'unable_to_determine_installed_plugin', __( 'Unable to determine what plugin was installed.', 'gutenberg' ), array( 'status' => 500 ) );
+		}
+
+		if ( 'inactive' !== $request['status'] ) {
+			$can_change_status = $this->plugin_status_permission_check( $file, $request['status'], 'inactive' );
+
+			if ( is_wp_error( $can_change_status ) ) {
+				return $can_change_status;
+			}
+
+			$changed_status = $this->handle_plugin_status( $file, $request['status'], 'inactive' );
+
+			if ( is_wp_error( $changed_status ) ) {
+				return $changed_status;
+			}
+		}
+
+		$path          = WP_PLUGIN_DIR . '/' . $file;
+		$data          = get_plugin_data( $path, false, false );
+		$data['_file'] = $file;
+
+		$response = $this->prepare_item_for_response( $data, $request );
+		$response->set_status( 201 );
+		$response->header( 'Location', rest_url( $this->namespace . '/' . $this->rest_base . '/' . substr( $file, 0, - 4 ) ) );
+
+		return $response;
+	}
+
+	/**
+	 * Checks if a given request has access to update a specific plugin.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to update the item, WP_Error object otherwise.
+	 */
+	public function update_item_permissions_check( $request ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		if ( ! current_user_can( 'activate_plugins' ) ) {
```

Comment:
What about network activating a plugin?

Reply:
`activate_plugins` is used as a barrier to all plugin management, that's how `wp-admin/plugins.php` works. The status checks for activating/deactivating are in `plugin_status_permission_check` which has a check for `network-active` plugins.